src/marqo/core/index_management/index_management.py

```diff
@@ -86,15 +86,24 @@ def bootstrap_vespa(self) -> bool:
         Returns:
             True if Vespa was bootstrapped, False if it was already up-to-date
         """
-        with self._vespa_deployment_lock():
-            vespa_app = self._get_vespa_application(check_configured=False, need_binary_file_support=True)
 
-            to_version = version.get_version()
-            from_version = vespa_app.get_marqo_config().version if vespa_app.is_configured else None
+        # We skip the Vespa convergence check here so that Marqo instance can be bootstrapped even when Vespa is
+        # not converged. This will reduce the risk of having downtime when the Marqo instance crashes while waiting
+        # for Vespa app to converge.
+        to_version = version.get_version()
+        vespa_app_for_version_check = self._get_vespa_application(check_configured=False, need_binary_file_support=True,
+                                                                  check_for_application_convergence=False)
+        from_version = vespa_app_for_version_check.get_marqo_config().version \
+            if vespa_app_for_version_check.is_configured else None
+
+        if from_version and semver.VersionInfo.parse(from_version) >= semver.VersionInfo.parse(to_version):
+            # skip bootstrapping if already bootstrapped to this version or later
+            return False
 
-            if from_version and semver.VersionInfo.parse(from_version) >= semver.VersionInfo.parse(to_version):
-                # skip bootstrapping if already bootstrapped to this version or later
-                return False
+        with self._vespa_deployment_lock():
+            # Initialise another session based on the current active session
+            vespa_app = self._get_vespa_application(check_configured=False, need_binary_file_support=True,
+                                                    check_for_application_convergence=False)
```

Comment:
Did we decide that the convergence check was superfluous anyway? Is there a scenario where we actually want this check? I would've thought it would reduce the chance of a concurrency failure if we did the check here

Reply:
Also what happens if a convergence is in progress but we go and deploy anyway before it converges? Does the deployment go through or fail?